### PR TITLE
Introduce basic pre/post processor logic

### LINKED ISSUE #93: [Feature] Post-Processor
Be able to pass in a function to modify the output before stitching and sending it to the external log function.

The default should be `null` with an if statement to prevent unnecessary function callings.

### PR DESCRIPTION
**Pull request by Antony1060**
Closes #93

### CODE REVIEW

**Comment by Antony1060** [comment]
Need to fix test coverage, it has some weird artifacts

**Review by svemat01 — APPROVED** [review]
LGTM but wait for another review

File: src/index.ts [hunk 1]
```diff
@@ -72,6 +72,10 @@ export type LogConfig = SharedConfig & {
      * @example ['error', 'important', 'success'] (only logs error, important, and success)
      */
     filter: RuntimeOrValue<string[] | undefined>;
+
```
Comment (lucemans) [inline src/index.ts:75]: JSDocs?

File: src/index.ts [hunk 2]
```diff
@@ -72,6 +72,10 @@ export type LogConfig = SharedConfig & {
...
+    preProcessors?: ((input: LogMethodInput[], method: { name: M } & MethodConfig, logger: LogConfig<M>) => LogMethodInput[])[];
+
```
Comment (lucemans) [inline src/index.ts:77]: JSDocs?

**Review by lucemans — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (Antony1060) [reply src/index.ts:75]: Sowwy

**Review by lucemans — APPROVED** [review]
LGTM

**Pull request merged**